script/model.py

```diff
@@ -4,141 +4,350 @@
-        self.belief_to_state_mean = nn.Sequential(
-            nn.Linear(self.belief_state_size, self.state_size),
-            nn.ReLU(),
-            nn.Linear(self.state_size, self.state_size))
-
-        self.belief_to_state_logstd = nn.Sequential(
-            nn.Linear(self.belief_state_size, self.state_size),
-            nn.ReLU(),
-            nn.Linear(self.state_size, self.state_size))               
-        
-        ## state to observation
-        self.state_to_obs = nn.Sequential(
-            nn.Linear(self.state_size, self.state_size),
-            nn.ReLU(),
-            nn.Linear(self.state_size, self.input_size),
-            nn.Sigmoid())        
-
-        ## state transition
-        self.state_to_state_mean = nn.Sequential(
-            nn.Linear(self.state_size, self.state_size),
-            nn.ReLU(),
-            nn.Linear(self.state_size, self.state_size))
-        
-        self.state_to_state_logstd = nn.Sequential(
-            nn.Linear(self.state_size, self.state_size),
-            nn.ReLU(),
-            nn.Linear(self.state_size, self.state_size))
-
-        ## inference state
-        self.infer_state_mu = nn.Sequential(
-            nn.Linear(self.belief_state_size*2 + self.state_size, self.state_size),
-            nn.ReLU(),
-            nn.Linear(self.state_size, self.state_size))
-
-        self.infer_state_logstd = nn.Sequential(
-            nn.Linear(self.belief_state_size*2 + self.state_size, self.state_size),
-            nn.ReLU(),
-            nn.Linear(self.state_size, self.state_size))        
-        
+        self.x_size = x_size
+        self.processed_x_size = processed_x_size
+        self.b_size = b_size
+        self.z_size = z_size
+
+        # input pre-processing layer
+        self.process_x = PreProcess(self.x_size, self.processed_x_size)
+
+        # one layer LSTM for aggregating belief states
+        # One layer LSTM is used here and I am not sure how many layers
+        # are used in the original paper
+
+        # batch_first – If True, then the input and output tensors
+        # are provided as (batch, seq, feature) instead of (seq, batch,
+        # feature)
+        self.lstm = nn.LSTM(input_size=self.processed_x_size,
+                            hidden_size=self.b_size,
+                            batch_first=True)
+
+        # Two layer state model is used. Sampling is done by sampling
+        # higher layer first.
+        # belief to state (b to z)
+
+        # (this is corresponding to probability of belief distribution in the
+        # reference; weights are shared across time but not across layers.)
+
+        # DBlock: parametralize a normal distribution
+        # input_size, hidden_size, output_size
+        # hidden layer size = 50 from original paper
+        # b_size = 8
+        # z_size = 50
+        self.l2_b_to_z = DBlock(b_size, 50, z_size)  # layer 2
+        ######### why does input size = b_size + z_size????
```

Comment:
to clean

Reply:
still confused about a lot of these dimensions. Maybe since you've had time to look at the code we can talk through it a little bit more.